src/crc16/simd.rs

```diff
@@ -0,0 +1,74 @@
+use crate::simd::{crc32_simd_coefficients, crc32_update_refin, SimdValue};
+use crate::table::crc16_table_slice_16;
+use crate::{Algorithm, Crc, Digest, Simd};
+
+use super::{finalize, init, update_slice16};
+
+impl Crc<Simd<u16>> {
+    pub const fn new(algorithm: &'static Algorithm<u16>) -> Self {
+        Self {
+            algorithm,
+            table: (
+                crc16_table_slice_16(algorithm.width, algorithm.poly, algorithm.refin),
+                unsafe {
+                    // SAFETY: Both represent numbers.
+                    core::mem::transmute(crc32_simd_coefficients(
+                        algorithm.width,
+                        algorithm.poly as u32,
+                    ))
+                },
+            ),
+        }
+    }
+
+    pub fn checksum(&self, bytes: &[u8]) -> u16 {
+        let mut crc = init(self.algorithm, self.algorithm.init);
+        crc = self.update(crc, bytes);
+        finalize(self.algorithm, crc)
+    }
+
+    fn update(&self, mut crc: u16, bytes: &[u8]) -> u16 {
+        if !self.algorithm.refin {
+            return update_slice16(crc, self.algorithm.refin, &self.table.0, bytes);
+        }
+
+        // SAFETY: Both represent numbers.
```

Comment:
Is it worth using [`bytemuck`](https://docs.rs/bytemuck/latest/bytemuck/fn.pod_align_to.html) here? It supports SIMD types: https://docs.rs/bytemuck/latest/bytemuck/trait.Pod.html#impl-Pod-for-__m128i.